server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java

```diff
@@ -75,10 +92,111 @@ protected int run(String[] args) {
     if (MODE_START.equals(mode)) {
       dataNode.doAddNode(args);
     } else if (MODE_REMOVE.equals(mode)) {
-      dataNode.doRemoveNode(args);
+      doRemoveNode(args);
     } else {
       logger.error("Unrecognized mode {}", mode);
     }
     return 0;
   }
+
+  /**
+   * remove datanodes from cluster
+   *
+   * @param args IPs for removed datanodes, split with ','
+   */
+  private void doRemoveNode(String[] args) {
+    try {
+      removePrepare(args);
+      removeNodesFromCluster(args);
+      removeTail();
+    } catch (Exception e) {
+      logger.error("remove Data Nodes error", e);
+    }
+  }
+
+  private void removePrepare(String[] args) throws BadNodeUrlException {
+    ConfigNodeInfo.getInstance()
+        .updateConfigNodeList(IoTDBDescriptor.getInstance().getConfig().getTargetConfigNodeList());
```

Comment:
Is it necessary to updateConfigNodeList here?

Reply:
necessary